homeassistant/components/peco/sensor.py

```diff
@@ -90,20 +104,28 @@ class PecoSensor(CoordinatorEntity, SensorEntity):
     def __init__(
         self,
         hass: HomeAssistant,
-        name: str,
+        description: SensorEntityDescription,
         county: str,
         coordinator: DataUpdateCoordinator,
-        key: str,
     ) -> None:
         """Initialize the sensor."""
         super().__init__(coordinator)
         self.hass = hass
         self._county = county
-        self._attr_name = name
-        self._attr_unique_id = f"{self._county}_{key}"
-        self._key = key
+        self._attr_name = f"{county.capitalize()} {description.name}"
+        self._attr_unique_id = f"{self._county}_{description.key}"
+        self._key = description.key
+        self._unit_of_measurement = description.native_unit_of_measurement
 
     @property
     def native_value(self) -> int:
         """Return the value of the sensor."""
         return self.coordinator.data[self._key]
+
+    @property
+    def native_unit_of_measurement(self) -> Any:
+        """Return the unit of measurement."""
+        if self._unit_of_measurement is not None:
+            return self._unit_of_measurement
+        else:
+            return None
```

Comment:
`this is the same as `return self._unit_of_measurement` - so you can just as well use `self._attr_unit_of_measurement` or just omit it because the SensorEntityDescription unit will be used anyway.